drivers/SmartThings/zigbee-switch/src/test/test_multi_switch_no_master.lua

```diff
@@ -19,184 +19,319 @@ local capabilities = require "st.capabilities"
 local OnOff = clusters.OnOff
 local zigbee_test_utils = require "integration_test.zigbee_test_utils"
 local t_utils = require "integration_test.utils"
+local profile = t_utils.get_profile_definition("multi-switch-no-master.yml")
 
-local mock_device = test.mock_device.build_test_zigbee_device(
+
+local mock_parent_device = test.mock_device.build_test_zigbee_device(
   {
-    profile = t_utils.get_profile_definition("multi-switch-no-master-2.yml"),
+    profile = profile,
     zigbee_endpoints = {
       [1] = {
         id = 1,
         manufacturer = "DAWON_DNS",
-        model = "PM-S240-ZB",
+        model = "PM-S340R-ZB",
         server_clusters = {}
       }
-    }
+    },
+    fingerprinted_endpoint_id = 0x01
   }
 )
 
+local mock_first_child = test.mock_device.build_test_child_device({
+  profile = profile,
+  device_network_id = string.format("%04X:%02X", mock_parent_device:get_short_address(), 2),
+  parent_device_id = mock_parent_device.id,
+  parent_assigned_child_key = string.format("%02X", 2)
+})
+
+local mock_second_child = test.mock_device.build_test_child_device({
+  profile = profile,
+  device_network_id = string.format("%04X:%02X", mock_parent_device:get_short_address(), 3),
+  parent_device_id = mock_parent_device.id,
+  parent_assigned_child_key = string.format("%02X", 3)
+})
+
 zigbee_test_utils.prepare_zigbee_env_info()
 local function test_init()
-  test.mock_device.add_test_device(mock_device)
-  test.mock_device.add_test_device(mock_device)
+  test.mock_device.add_test_device(mock_parent_device)
+  test.mock_device.add_test_device(mock_first_child)
+  test.mock_device.add_test_device(mock_second_child)
   zigbee_test_utils.init_noop_health_check_timer()
 end
 
 test.set_test_init_function(test_init)
 
 test.register_message_test(
-  "Reported on off status should be handled: on ep 1",
+  "Refresh on parent device should read all necessary attributes",
+  {
+    {
+      channel = "capability",
+      direction = "receive",
+      message = {mock_parent_device.id, {capability = "refresh", component = "main", command = "refresh", args = {}}}
```

Comment:
nit: space after opening bracket is missing

Reply:
done